File: myExpenses/src/main/java/org/totschnig/myexpenses/dialog/GrisbiSourcesDialogFragment.java
package org.totschnig.myexpenses.dialog;

import android.content.DialogInterface;
import android.view.View;

import org.totschnig.myexpenses.R;
import org.totschnig.myexpenses.activity.GrisbiImport;

import androidx.appcompat.app.AlertDialog;

public class GrisbiSourcesDialogFragment extends TextSourceDialogFragment implements
DialogInterface.OnClickListener {
  
  public static GrisbiSourcesDialogFragment newInstance() {
    return new GrisbiSourcesDialogFragment();
  }
  @Override
  protected int getLayoutId() {
    return R.layout.grisbi_import_dialog;
  }
  @Override
  protected String getLayoutTitle() {
    return getString(R.string.pref_import_from_grisbi_title);
  }

  @Override
  public String getTypeName() {
    return "Grisbi XML";
  }
  @Override
  public String getPrefKey() {
    return "import_grisbi_file_uri";
  }

  @Override
  public void onClick(DialogInterface dialog, int id) {
    if (getActivity()==null) {
      return;
    }
    if (id == AlertDialog.BUTTON_POSITIVE) {
      maybePersistUri();
      ((GrisbiImport) getActivity()).onSourceSelected(
          mUri,
          mImportCategories.isChecked(),
          mImportParties.isChecked()
          );
    } else {
      super.onClick(dialog, id);
    }
  }
  @Override
  protected void setupDialogView(View view) {
    super.setupDialogView(view);
    mImportTransactions.setVisibility(View.GONE);
  }
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/dialog/GrisbiSourcesDialogFragment.kt
package org.totschnig.myexpenses.dialog

import android.content.DialogInterface
import android.view.View
import androidx.appcompat.app.AlertDialog
import org.totschnig.myexpenses.R
import org.totschnig.myexpenses.activity.GrisbiImport

class GrisbiSourcesDialogFragment : TextSourceDialogFragment(), DialogInterface.OnClickListener {
    override val layoutId = R.layout.grisbi_import_dialog
    override val layoutTitle: String
        get() = getString(R.string.pref_import_from_grisbi_title)

    override val typeName = "Grisbi XML"

    override val prefKey = "import_grisbi_file_uri"

    override fun onClick(dialog: DialogInterface, id: Int) {
        if (activity == null) {
            return
        }
        if (id == AlertDialog.BUTTON_POSITIVE) {
            maybePersistUri()
            (activity as GrisbiImport?)!!.onSourceSelected(
                uri,
                mImportCategories.isChecked,
                mImportParties.isChecked
            )
        } else {
            super.onClick(dialog, id)
        }
    }

    override fun setupDialogView(view: View) {
        super.setupDialogView(view)
        mImportTransactions.visibility = View.GONE
    }

    companion object {
        @JvmStatic
        fun newInstance(): GrisbiSourcesDialogFragment {
            return GrisbiSourcesDialogFragment()
        }
    }
}